### PR TITLE
Allow including files just before manager.bundle.js

### PR DESCRIPTION
**Pull request by enjoylife**
## What I did
I simply followed the preexisting approach of including additional html, but instead of it being injected into the story iframe, it is injected in the index.html, right before the manager script.

This commit really was just to satisfy my issue detailed below, but the pull request is more an example for the maintainers to help solve a few outstanding issues/requests. See #678 , #686, #630.

## Context & Why?
The process went like this:
1. Why are storybook's build times so slow (5 sec or more)?
2. Lets try adding the webpack dll plugin
3. Oh wait, now I need to include a vendor.js (it has the react bits) prior to storybook
4. This commit
5. Now build times are sub 200ms.

## How to test
Include a file called `bodyscript.html` in the storybook config directory, and you will see it be injected into the source.

### CODE REVIEW

**Comment by ndelangen** [comment]
Is related to https://github.com/storybooks/storybook/pull/1083 I think..

Would it be an option to use DLLplugin by default? I haven't used it before, So I could use help with that.

**Comment by enjoylife** [comment]
From my limited knowledge (reading an hour and [this blog post](https://robertknight.github.io/posts/webpack-dll-plugins/)), with the DLL plugin, its a two step thing.
1. You have a separate webpack config which does just does the bundling of your "vendor" packages upfront.
 ` webpack --config .\.storybook\vender.webpack.config.js`
```js

//  .storybook/vendor.webpack.config.js

var path = require("path");
var webpack = require("webpack");

module.exports = {
  context: process.cwd(),
  entry: {
    vendor:[
      'react',
      'react-dom',
      'react-router'
       // any other "expensive" packages to build
      ]
  },

  output: {
    filename: '[name].dll.js',
    path: path.join(__dirname,'public'),
    library: '[name]',
  },

  plugins: [
    new webpack.DllPlugin({
      path: path.join(__dirname,'public', "[name]-manifest.json"),
      name: "[name]",
      context: path.resolve(__dirname)
    })
  ]
};
```
That outputs two files
```
public\vendor.dll.js
public\vendor-manifest.json
```
Then in the webpack config being used by storybook, we incorporate the run time part which doesn't output chunks but reads in the generated manifest

```
// .storybook/webpack.config.js
// ...
 plugins: [
    // Use the DLL in development.
    new webpack.DllReferencePlugin({
      context: __dirname,
      manifest: require("./public/vendor-manifest.json"),
    }) 
  ]
// ...
```
Then any chunks at runtime will be expecting the pregenerated vendor script to be loaded prior to their execution.

```
 <script src="vendor.dll.js"></script>
 <script src="static/manager.bundle.js"></script>
```
 So to answer one of your questions, It is related to #1083, but instead of dynamic chunks, its more about needing an additional explicit configurable list of public js files

**Comment by ndelangen** [comment]
Right, that confirms what I thought it was. This will definitely be supported one way or another!

Thank you so much for this crazy good PR! ❤️

**Comment by shilman** [comment]
> Now build times are sub 200ms.
> _Drops mic._

This is amazing. Thanks for this wonderful PR. 🙇

**Comment by ndelangen** [comment]
@enjoylife I'm going to investigate adding / using the DLL plugin to speed up the builds for everyone!

concerning this PR. What I think I'd like to do merge this, with a simple change.

Right now your proposal is to add support for a `bodyscript.html` file.
We already have a `head.html` file. As a consumer, I'd be really confused what file has what purpose.. What do you think of this suggestion:

- rename `head.html` to `preview-head.html` *
- add support for a new file called `manager-head.html`

> (*) if this file would not be found we would look for `head.html` for backwards compatibility.
> We could display a deprecation warning if `head.html` was provided, asking the user to rename it.

**Review by ndelangen — CHANGES_REQUESTED** [review]
Right now your proposal is to add support for a `bodyscript.html` file.
We already have a `head.html` file. As a consumer, I'd be really confused what file has what purpose.. What do you think of this suggestion:

- rename `head.html` to `preview-head.html` *
- add support for a new file called `manager-head.html`

> (*) if this file would not be found we would look for `head.html` for backwards compatibility.
> We could display a deprecation warning if `head.html` was provided, asking the user to rename it.

**Comment by enjoylife** [comment]
The original names in the pull request were hastily thought out... so if those changes better align with the existing naming conventions in storybook, then that is great!

**Comment by ndelangen** [comment]
I'm can accept this PR if you change the filenames (with support for _legacy_ `head.html`.

Bonuspoints for adding this behavior to the docs! ❤️

**Comment by ndelangen** [comment]
@enjoylife Are you interested in updating this PR? Is there something I can do to help?

**Comment by enjoylife** [comment]
@ndelangen, I updated the naming conventions as per our previous comments and added a few updates to the docs.

**Comment by shilman** [comment]
@ndelangen I think you are doing something funky with `git` here, and it is causing the merge commit to be improperly labeled, which is breaking my release notes generator. Please merge PR's from the github website unless you have a very good reason not to. For now I'll add this one by hand.

storybook_ _less_ _git_log_ _89x43

File: app/react/src/server/index.html.js [hunk 1]
```diff
@@ -70,6 +70,7 @@ export default function(data) {
             background-color: #eee
           }
         </style>
+        ${headHtml}
```
Comment (shilman) [inline app/react/src/server/index.html.js]: I think this is buggy when `headHtml` is not defined

**Pull request merged**